src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs

```diff
@@ -11,50 +12,124 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Refactor
 {
-    [Export(typeof(IImplicitlyActiveService))]
     [Export(typeof(IRefactorNotifyService))]
     [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
-    internal sealed partial class VisualStudioRefactorNotifyService : AbstractMultiLifetimeComponent<VisualStudioRefactorNotifyService.Instance>, IImplicitlyActiveService, IRefactorNotifyService
+    internal class VisualStudioRefactorNotifyService : IRefactorNotifyService
     {
-        private readonly IVsService<SDTE, DTE> _dte;
-        private readonly IVsService<SVsSolution, IVsSolution> _solutionService;
+        private readonly IVsUIService<DTE> _dte;
+        private readonly IVsUIService<IVsSolution> _solution;
 
         [ImportingConstructor]
-        public VisualStudioRefactorNotifyService(UnconfiguredProject unconfiguredProject,
-                                                 IProjectThreadingService threadingService,
-                                                 IVsService<SDTE, DTE> dte,
-                                                 IVsService<SVsSolution, IVsSolution> solutionService)
-            : base(threadingService.JoinableTaskContext)
+        public VisualStudioRefactorNotifyService(IVsUIService<SDTE, DTE> dte, IVsUIService<SVsSolution, IVsSolution> solution)
         {
             _dte = dte;
-            _solutionService = solutionService;
+            _solution = solution;
         }
 
-        public Task ActivateAsync() => LoadAsync();
-        public Task DeactivateAsync() => UnloadAsync();
+        public void OnBeforeGlobalSymbolRenamed(string projectPath, IEnumerable<string> filePaths, string rqName, string newName)
+        {
+            IVsHierarchy? projectHierarchy = GetProjectHierarchy(projectPath);
+            if (projectHierarchy == null)
+            {
+                return;
+            }
+
+            if (!(projectHierarchy is IVsHierarchyRefactorNotify refactorNotify))
```

Comment:
couldn't this just be a single if with an `||`?

Reply:
I didn't change this code.